packages/react-aria-components/src/FileTrigger.tsx

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2023 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {AriaLabelingProps} from '@react-types/shared';
+import {ContextValue, DOMProps, SlotProps, useContextProps} from './utils';
+import {filterDOMProps} from '@react-aria/utils';
+import {Input} from './Input';
+import {PressResponder} from '@react-aria/interactions';
+import React, {createContext, ForwardedRef, forwardRef, useRef} from 'react';
+import {VisuallyHidden} from 'react-aria';
+
+export interface FileTriggerProps extends SlotProps, DOMProps, AriaLabelingProps {
+  /**
+   * Specifies what mime type of files are allowed.
+   */
+  acceptedFileTypes?: Array<string>,
+  /**
+   * Whether multiple files can be selected.
+   */
+  allowsMultiple?: boolean,
+  /**
+   * Specifies the use of a media capture mechanism to capture the media on the spot.
+   */
+  defaultCamera?: 'user' | 'environment',
+  /**
+   * Handler when a user selects a file.
+   */
+  onChange?: (files: FileList | null) => void
```

Comment:
opinions on this type? got some lint errors saying `Argument of type 'FileList | null' is not assignable to parameter of type 'FileLists` so this was one solution but there are a couple of alternative solutions that we could try i think